### PR TITLE
Feat/#300 daco hook up

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

Adds DACO check for profile page 

**Type of Change**

- [ ] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

**Comment by ciaranschutte** [comment]
@hlminh2000 need to check with Kim about a loader

File: components/pages/user/index.tsx [hunk 1]
```diff
@@ -18,6 +20,10 @@ export function UserPage({ firstName, lastName }: { firstName: string, lastName:
       {...props}
     />
   );
+
+  const { data, loading } = useQuery(PROFILE);
```
Comment (hlminh2000) [inline components/pages/user/index.tsx]: maybe let's add a type for `data`? something like this should do :)
```
type ProfileQueryData = {
  self: {
    isDacoApproved: boolean
    apiKey: AccessKey
  }
}
```
then
```
useQuery<ProfileQueryData>(PROFILE)
```

Comment (ciaranschutte) [reply components/pages/user/index.tsx]: nice catch

**Comment by ciaranschutte** [comment]
@hlminh2000 added typing

**Comment by ciaranschutte** [comment]
@hlminh2000 addressed thank you :)

**Pull request merged**